### PR TITLE
benchmarks: add script to aggregate results from result_analyzer

### PR DESCRIPTION
**Pull request by cota**
See the description in the `benchmarks: add aggregate.py` commit.

### CODE REVIEW

**Review by frgossen — APPROVED** [review]
Nice! Thanks!

File: benchmarks/aggregate.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+def pr_latest(results_map: dict[str, Any], args, timestamps: list[str]):
+  label = f'speedups:{args.test}'
+  model_label = f'{label}:model_name'
+
+  for timestamp in reversed(timestamps):
+    if label not in results_map[timestamp][args.accelerator]:
+      continue
+    acc_map = results_map[timestamp][args.accelerator]
+    (speedups, model_names) = map(list, zip(*sorted(zip(acc_map[label], acc_map[model_label]))))
```
Comment (frgossen) [inline benchmarks/aggregate.py:180]: nit: these brackets aren't needed. Think it's idiomatic not to have them

File: benchmarks/aggregate.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        acc_map[model_label] = []
+      acc_map[model_label].append(model_name)
+  summarize_speedups(acc_map, label)
+
+
+def process_results(results_map: dict[str, Any]):
+  for timestamp in results_map:
+    for accelerator in results_map[timestamp]:
+      acc_map = results_map[timestamp][accelerator]
+
```
Comment (frgossen) [inline benchmarks/aggregate.py:159]: nit: intended empty line?

File: benchmarks/tiers.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+_FILTER_BY_TIER = {
```
Comment (frgossen) [inline benchmarks/tiers.py:1]: Nice to have them separate

File: benchmarks/aggregate.py (hunk 1 above)
Comment (cota) [reply benchmarks/aggregate.py]: Thanks, eventually done in https://github.com/pytorch/xla/pull/6062

**Pull request merged**